dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,537 @@
+     * <ul>
+     * <li>Information from the user: ID, associated company.</li>
+     * <li>The "Remember Me" option being checked or not.</li>
+     * </ul>
+     * The JWT will allow the user to access the dotCMS back-end even though their session has
+     * expired. This way, they won't need to re-authenticate.
+     *
+     * @param req - The {@link HttpServletRequest} object.
+     * @param res - The {@link HttpServletResponse} object.
+     * @param user - The {@link User} trying to log into the system.
+     * @param maxAge - The maximum days (in milliseconds) that the JWT will live. If the "Remember
+     *        Me" option is checked, the token will live for many days (check its default value).
+     *        Otherwise, it will only live during the current session.
+     * @throws PortalException The specified user could not be found.
+     * @throws SystemException An error occurred during the user ID encryption.
+     */
+    private void processJsonWebToken(final HttpServletRequest req, final HttpServletResponse res,
+                    final User user, final int maxAge) throws PortalException, SystemException {
+
+        final String jwtAccessToken = this.jsonWebTokenUtils.createToken(user, maxAge);
+        createJsonWebTokenCookie(req, res, jwtAccessToken, Optional.of(maxAge));
+    }
+
+
+
+    @Override
+    public void doRememberMe(final HttpServletRequest req, final HttpServletResponse res,
+                    final User user, final boolean rememberMe) {
+
+        int jwtMaxAge = rememberMe ? Config.getIntProperty(JSON_WEB_TOKEN_DAYS_MAX_AGE,
+                        JSON_WEB_TOKEN_DAYS_MAX_AGE_DEFAULT) : -1;
+
+        this.doRememberMe(req, res, user, jwtMaxAge);
+    } // doRememberMe.
+
+    @Override
+    public void doRememberMe(final HttpServletRequest req, final HttpServletResponse res,
+                    final User user, final int maxAge) {
+
+        try {
+
+            this.processJsonWebToken(req, res, user, maxAge);
+        } catch (Exception e) {
+
+            Logger.debug(this, e.getMessage(), e);
+        }
+    } // doRememberMe.
+
+
+    private boolean _doCookieLogin(String encryptedId, HttpServletRequest request,
+                    HttpServletResponse response) {
+
+        try {
+            String decryptedId = PublicEncryptionFactory.decryptString(encryptedId);
+            /* Custom Code */
+            User user = null;
+            if (Validator.isEmailAddress(decryptedId))
+                user = userAPI.loadByUserByEmail(decryptedId, systemUser, false);
+            else
+                user = userAPI.loadUserById(decryptedId, systemUser, false);
+            /* End of Custom Code */
+            try {
+                String userName = (authByEmail) ? user.getEmailAddress() : user.getUserId();
+
+                return _doLogin(userName, null, true, request, response, true);
+            } catch (Exception e) { // $codepro.audit.disable logExceptions
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)